src/modules/Dashboard/features/Applet/Activities/Activities.tsx

```diff
@@ -1,18 +1,266 @@
+  }, [appletId]);
+
+  const isEmpty = !activities?.length && !isLoading;
+
+  const getEmptyComponent = () => {
+    if (isEmpty) {
+      if (searchValue) {
+        return t('noMatchWasFound', { searchValue });
+      }
+
+      return appletId ? t('noActivitiesForApplet') : t('noActivities');
+    }
+  };
+
+  return (
+    <StyledBody>
+      {isLoading && <Spinner />}
+
+      <StyledFlexWrap sx={{ gap: 1.2, mb: 2.4 }}>
+        {appletId && (
+          <>
+            <StyledFlexTopCenter sx={{ gap: 1.2 }}>
+              <StyledButton
+                variant="outlined"
+                startIcon={<Svg width={18} height={18} id="slider-rows" />}
+                // TODO: Implement filters
+                // https://mindlogger.atlassian.net/browse/M2-5530
+                onClick={() => alert('TODO: filters')}
+                data-testid={`${dataTestid}-filters`}
+              >
+                {t('filters')}
+              </StyledButton>
+
+              <ButtonWithMenu
+                variant="outlined"
+                label={t('sortBy')}
+                anchorEl={anchorEl}
+                setAnchorEl={setAnchorEl}
+                // TODO: Implement sorting
+                // https://mindlogger.atlassian.net/browse/M2-5445
+                menuItems={[
+                  {
+                    title: 'TODO: Sort options',
+                    action: () => alert('TODO: Sort options'),
+                  },
+                ]}
+                startIcon={<></>}
+                data-testid={`${dataTestid}-sort-by`}
+              />
+            </StyledFlexTopCenter>
+
+            <StyledFlexWrap sx={{ gap: 1.2, ml: 'auto' }}>
+              <StyledSearch
+                withDebounce
+                // TODO: Implement search
+                // https://mindlogger.atlassian.net/browse/MLG-21
+                placeholder={`TODO: ${t('searchActivities')}`}
+                onSearch={handleSearch}
+                data-testid={`${dataTestid}-search`}
+              />
+
+              <StyledButton
+                variant="outlined"
+                // TODO: Implement assign
+                // https://mindlogger.atlassian.net/browse/M2-5710
+                onClick={() => alert('TODO: Assign activity')}
+                data-testid={`${dataTestid}-assign`}
+                sx={{ minWidth: '10rem' }}
+              >
+                {t('assign')}
+              </StyledButton>
+
+              <StyledButton
+                variant="contained"
+                onClick={() => navigate(generatePath(page.builderAppletActivities, { appletId }))}
+                data-testid={`${dataTestid}-add-activity`}
+                sx={{ minWidth: '13.2rem' }}
+              >
+                {t('addActivity')}
+              </StyledButton>
+            </StyledFlexWrap>
+          </>
+        )}
+      </StyledFlexWrap>
+
+      {!!activities.length && (
+        <StyledFlexWrap sx={{ gap: 2.4 }} data-testid={`${dataTestid}-grid`}>
```

Comment:
Even though they're styled similarly, the scroll behavior of the activities tab and the participants tab end up being different because on the participants tab the table stretches to fill the available space. That means that on the activities page the control buttons at the top end up scrolling off-screen.

https://github.com/ChildMindInstitute/mindlogger-admin/assets/14842108/09beeec9-e1d3-46e9-80bb-ce698e430e72

I'm not sure what the intended behavior is (it's not clear from the designs), but I think keeping them visible would be a good choice. What do you think?

https://github.com/ChildMindInstitute/mindlogger-admin/assets/14842108/d1b265ca-7d6f-4110-94fe-ae27938bbe18

```suggestion
        <StyledFlexWrap sx={{ gap: 2.4, overflowY: 'visible' }} data-testid={`${dataTestid}-grid`}>
```

Reply:
Absolutely! Good call